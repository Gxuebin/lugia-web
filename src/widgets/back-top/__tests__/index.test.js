/**
 *
 * create by ligx
 *
 * @flow
 */
import React from 'react';
import chai from 'chai';

import 'jest-styled-components';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import BackTopDemo from '../demo';
import BackTop from '../back-top';
import renderer from 'react-test-renderer';
const { expect: exp } = chai;

Enzyme.configure({ adapter: new Adapter() });

describe('BackTop', () => {
  it('css', () => {
    const Target = <BackTopDemo />;
    expect(renderer.create(Target).toJSON()).toMatchSnapshot();
  });
  it('BackTop needFixed ', () => {
    const target = mount(<BackTop />);
    const backTopElement = target
      .children()
      .at(0)
      .instance();

    expect(backTopElement.needFixed(100, 50)).toBe(true);
    expect(backTopElement.needFixed(50, 100)).toBe(false);
  });
});
