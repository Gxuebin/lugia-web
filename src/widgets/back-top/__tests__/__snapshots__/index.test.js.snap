// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackTop css 1`] = `
.c0 > span > div > div {
  bottom: 100px;
}

.c1 > span > div > div {
  bottom: 150px;
}

.c2 > span > div > div {
  bottom: 200px;
}

<div
  style={
    Object {
      "height": "1600px",
    }
  }
>
  <span>
    <div />
  </span>
  <div
    className="c0"
  >
    <span>
      <div />
    </span>
  </div>
  <div
    className="c1"
  >
    <span>
      <div />
    </span>
  </div>
  <div
    style={
      Object {
        "height": "200px",
        "marginTop": "400px",
        "overflowY": "scroll",
        "width": "200px",
      }
    }
  >
    <div
      className="c2"
    >
      <div
        style={
          Object {
            "height": "300px",
            "width": "20px",
          }
        }
      />
      <span>
        <div />
      </span>
    </div>
  </div>
</div>
`;
